go-controller/pkg/cni/types/types.go

```diff
@@ -9,10 +9,14 @@ import (
 // NetConf is CNI NetConf with DeviceID
 type NetConf struct {
 	types.NetConf
-	// PrimaryNetwork is valid on only L3 / L2 topologies. Not on localnet.
-	// It allows for using this secondary network as the primary network
-	// for the pod in order to achieve native network segmentation
-	PrimaryNetwork bool `json:"primaryNetwork,omitempty"`
+	// Role is valid only on L3 / L2 topologies. Not on localnet.
+	// It allows for using this network to be either secondary or
+	// primary user defined network for the pod.
+	// primary user defined networks are used in order to achieve
+	// native network isolation.
+	// In order to ensure backwards compatibility, if empty the
+	// network is considered secondary
+	Role string `json:"role,omitempty"`
```

Comment:
Should we mention "default" at the end internally it has a NetConf too.

Reply:
I think given this is a user facing struct we probably don't want to mention default anywhere here.. ? let's keep it only on primary/secondary being the values for user facing